src/main/java/com/microsoft/alm/authentication/KeychainSecurityCliStore.java

```diff
@@ -304,6 +307,22 @@ else if ("blob".equals(type.toString()))
         // TODO: else if ("sint32".equals(type))
     }
 
+    public boolean isKeychainAvailable()
+    {
+        final String targetName = "test:isKeychainAvailable";
+        final Token token = new Token("this is a test token", TokenType.Test);
+        try
+        {
+            writeToken(targetName, token);
+        }
+        catch (final SecurityException e)
+        {
+            return false;
+        }
+        delete(targetName);
+        return true;
+    }
```

Comment:
In isKeychainAvailable(), if an exception is thrown during delete(targetName), you could return false rather than allowing the exception to bubble up.